### ISSUE TITLE
bug: PENPOT_OIDC_ROLES_ATTR does not work

### ISSUE THREAD

**Issue by wdenbakker** [uxbox/uxbox#3152]
### Steps To Reproduce

1. Start penpot with:
```
PENPOT_FLAGS=enable-login-with-oidc
PENPOT_OIDC_CLIENT_ID=penpot
PENPOT_OIDC_CLIENT_SECRET=qwerty123
PENPOT_OIDC_BASE_URI=https://keycloak.example.com/realms/myrealm/
PENPOT_OIDC_ROLES=penpot_user
PENPOT_OIDC_ROLES_ATTR=roles
```
2. Login to penpot using oidc with a user that has role 'penpot_user'.

### Expected behavior

Succesfully login if the user has the 'penpot_user' role.

### Actual behavior

Fails to login even if the user has the 'penpot_user' role.

### Backend Stack Trace

```console
[2023-04-20 12:33:18.345] E app.auth.oidc - hint="error on oauth process"
SUMMARY:
 →  clojure.lang.ExceptionInfo: not enough permissions (oidc.clj:395)
DETAIL:
 →  clojure.lang.ExceptionInfo: not enough permissions (oidc.clj:395)
    at: app.auth.oidc$get_info$validate_oidc__34348.invoke(oidc.clj:395)
        promesa.util.Function.apply(util.cljc:40)
        java.util.concurrent.CompletableFuture$UniApply.tryFire(CompletableFuture.java:646)
        java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:510)
        java.util.concurrent.CompletableFuture.postFire(CompletableFuture.java:614)
        java.util.concurrent.CompletableFuture$UniWhenComplete.tryFire(CompletableFuture.java:844)
        java.util.concurrent.CompletableFuture$Completion.exec(CompletableFuture.java:483)
        java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373)
        java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1182)
        java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1655)
        java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1622)
        java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:165)
    dt: {:type :internal, :code :unable-to-auth, :hint "not enough permissions"}
```

### Additional context

I can successfully login when not using PENPOT_OIDC_ROLES and PENPOT_OIDC_ROLES_ATTR, so the remaining configuration is at least correct.

Modifying the [error message](https://github.com/penpot/penpot/blob/ed164ce69b5f88bbb83f84e9b5ef7ea237f17960/backend/src/app/auth/oidc.clj#L400) to print the 'info' map results in:
```clj
{
  :backend "oidc",
  :email "john.snow@example.com",
  :fullname "John Snow",
  :props {
    :oidc/given_name "John",
    :oidc/sub "20309140-2692-42c2-bb2a-233421cdb3eb",
    :oidc/roles ["penpot_user"],
    :oidc/email_verified true,
    :oidc/family_name "Snow",
    :oidc/preferred_username "john_snow"
  }
}
```
From what I can [see](https://github.com/penpot/penpot/blob/ed164ce69b5f88bbb83f84e9b5ef7ea237f17960/backend/src/app/auth/oidc.clj#L389-L390) the PENPOT_OIDC_ROLES_ATTR selects the top-level key of the info object, not a key nested in 'props', making it impossible to use any user key. Changing it to `roles (get (get info :props) :oidc/roles)]` does result in a correct login for my use case.

**Comment by niwinz** [comment]
Looks like a bug, I will check it thanks

**Comment by niwinz** [comment]
Fixed, the fix will be released on 1.18.3 soon

**Comment by icsy7867** [comment]
@niwinz 

Will this also help with the email and name attribute mapping?  I have a similar use case where my email and name attributes are nested, and my IT organizations says this works fine for everyone else (Using the example here):

```
{
  :backend "oidc",
  :email "",
  :fullname "",
  :props {
    :oidc/given_name "John",
    :oidc/sub "20309140-2692-42c2-bb2a-233421cdb3eb",
    :oidc/roles ["penpot_user"],
    :oidc/email_verified true,
    :oidc/family_name "Snow",
    :oidc/preferred_username "john_snow",
    :oidc/attributes {
      :email "john.snow@example.com",
      :name "last, first",
      :family_name "Smith",
      :preferred_username "jsmith"
    }
  }
}
```

So I would need to go:
props -> OIDC/Attributes -> email
props -> OIDC/Attributes -> name

**Comment by niwinz** [comment]
Hmm, no, but I'm in the time to improve it, so I will do it.

**Comment by niwinz** [comment]
Yep, an update:

I have normalized all props lookup... so for the data structure that you have mentioned, the env variables should be:

```
export PENPOT_OIDC_ROLES_ATTR=roles
export PENPOT_OIDC_EMAIL_ATTR=attributes__email
export PENPOT_OIDC_NAME_ATTR=attributes__name
```

This change will be released with 1.8.3

**Comment by icsy7867** [comment]
PENPOT_OIDC_EMAIL_ATTR and PENPOT_OIDC_NAME_ATTR nested values working beautifully for me! Thank you.
